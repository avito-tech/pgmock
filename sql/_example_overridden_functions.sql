create function public.foo() returns integer language plpgsql as
$func$
begin
    return 42;
end;
$func$;

create function public.foo(param integer) returns integer language plpgsql as
$func$
begin
    return param;
end;
$func$;

select pgmock.mock($$"${'public.foo()'::regprocedure}"$$);

select public.foo() = pg_temp.foo();

select pgmock.mock($$"${'public.foo(integer)'::regprocedure}"$$);

select public.foo(8) = pg_temp.foo(8);