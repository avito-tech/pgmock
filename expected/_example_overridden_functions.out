create function public.foo() returns integer language plpgsql as
$func$
begin
    return 42;
end;
$func$;
create function public.foo(param integer) returns integer language plpgsql as
$func$
begin
    return param;
end;
$func$;
select pgmock.mock($$"${'public.foo()'::regprocedure}"$$);
 mock 
------
 
(1 row)

select public.foo() = pg_temp.foo();
 ?column? 
----------
 t
(1 row)

select pgmock.mock($$"${'public.foo(integer)'::regprocedure}"$$);
 mock 
------
 
(1 row)

select public.foo(8) = pg_temp.foo(8);
 ?column? 
----------
 t
(1 row)

